sdk/nodejs/provider/experimental/analyzer.ts

```diff
@@ -584,6 +588,26 @@ function isInput(type: typescript.Type): boolean {
     return hasOutput && hasPromise && hasOther;
 }
 
+function isResourceReference(type: typescript.Type, checker: typescript.TypeChecker): boolean {
+    if (type.isClassOrInterface()) {
+        const baseTypes = checker.getBaseTypes(type as typescript.InterfaceType);
+        for (const baseType of baseTypes) {
+            const symbol = baseType.getSymbol();
+            const matchesName =
+                symbol?.escapedName === "CustomResource" ||
+                symbol?.escapedName === "ComponentResource" ||
+                symbol?.escapedName === "Resource";
```

Comment:
We seem to have a different approach in https://github.com/pulumi/pulumi/blob/fabeb12b4f74b61ff44b4c87fadc174b89f0f47e/sdk/nodejs/provider/experimental/analyzer.ts#L202

Can they be unified?

Reply:
Not easily. At that point we're looking at the class definition, while here we're looking at the type and extract the information from that.  There might be a way to get the node of the class definition here, but just using the type seems simpler.